### DISCUSSION TITLE
Can't get it running

### DISCUSSION THREAD

**Discussion by paysan9136** [bpatrik/pigallery2#335] [General]
Hi Patrik,
PiGallery2 looks very great. Thanks for this and thanks for making it available to the world.
Therfore I want to have it up and running on my PI-4 running Buster and is USB bootable from a 500GB SSD.
I followed the steps installing it using Docker.
But I can't get it running.  
In my home directory there us a file named: docker-compose.yml
sofar, docker runs ok. The Hello-World test works.
Installed Node.js and mpn and nginx.

I went to the https://hub.docker.com/r/bpatrik/pigalleery2/tags/ and choosed latest-debian-buster.

root@pigallery:/home/pi# docker pull bpatrik/pigallery2:latest-debian-buster
latest-debian-buster: Pulling from bpatrik/pigallery2
Digest: sha256:21fbaa5fae618503783f41b384c2bb700aea6b52bda1d4fd682579a3e59abcc6
Status: Image is up to date for bpatrik/pigallery2:latest-debian-buster
docker.io/bpatrik/pigallery2:latest-debian-buster
root@pigallery:/home/pi# 

so that looks ok so far I think.

but when I type:

root@pigallery:/home/pi# npm start
npm ERR! code ENOENT
npm ERR! syscall open
npm ERR! path /home/pi/package.json
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, open '/home/pi/package.json'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-06-25T11_42_45_656Z-debug.log
root@pigallery:/home/pi# 

root@pigallery:/home/pi# more  /root/.npm/_logs/2021-06-25T11_42_45_656Z-debug.log
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'start' ]
2 info using npm@6.14.12
3 info using node@v12.22.1
4 verbose stack Error: ENOENT: no such file or directory, open '/home/pi/package.json'
5 verbose cwd /home/pi
6 verbose Linux 5.10.17-v8+
7 verbose argv "/usr/bin/node" "/usr/bin/npm" "start"
8 verbose node v12.22.1
9 verbose npm  v6.14.12
10 error code ENOENT
11 error syscall open
12 error path /home/pi/package.json
13 error errno -2
14 error enoent ENOENT: no such file or directory, open '/home/pi/package.json'
15 error enoent This is related to npm not being able to find a file.
16 verbose exit [ -2, true ]
root@pigallery:/home/pi# 

But how to start PiGallery2 ????

Please can someone help me out?

**Comment by bpatrik** [comment]
> Installed Node.js and mpn and nginx.
 you do not need those. the docker image should contain everything.

You need to run `docker-compose up -d` to start the app from the folder where the compose file is, see documentation: 
https://github.com/bpatrik/pigallery2/tree/master/docker#i1b-start-docker-compose

**Reply by paysan9136** [reply]
Still no luck.
I did the pull and with “docker images” I can see the image is on my system. When I do “docker run latest-debian-buster” I get
Unable to find image ‘latest-debian-buster ’ locally.
docker: Error response from daemon: pull access denied etc….
Why is here a new pull request?  I can see with the docker images that the image is already on this system.

Why do you use ‘docker-compose up -d’ ?
I tried that as well but I can’t find the folder where the compose file is.

**Comment by bpatrik** [comment]
The image should be something like pigallery 2/latest-debian-buster, like
it is stated in the docker compose file.



-- Sorry for being brief, sent from my phone.

paysan9136 ***@***.***> ezt írta (időpont: 2021. jún. 28., H
20:39):

**Comment by paysan9136** [comment]
I got access denied when I use:
docker run bpatrik/gallery2/latest-debian-buster or
docker run gallery2/latest-debian/buster

Please help....
